eng/common/TestResources/remove-test-resources.yml

```diff
@@ -21,6 +21,8 @@ parameters:
 
 steps:
   - pwsh: |
+      . "eng/common/TestResources/Update-PSModulePathForHostedAgent.ps1"
```

Comment:
If we can make these persist in the environment then we should only need to do it in the setup steps and not here.

Reply:
Based on a quick review of the utility script it does look to be setting the `$env:PSModulePath` which is an environment variable but I guess the trouble is that it might not persist across steps. We can consider setting this as a devops variable so that it does persist for future steps.

Reply:
I considered this, but for now I wanted to keep things as close as possible to the `AzurePowerShell` task which only sets the module path for the current PowerShell invocation.

Reply:
Ok this can work but we will have to remember to do this in any steps that need to use the cmdlets.

Reply:
I think DevOps works this way intentionally, since different steps might need to use different versions of the cmdlets, and I think its difficult or impossible to load multiple versions side-by-side.

Reply:
OK I think we can live with that.

Given these are setting an env variable why are you dot-sourcing the script as opposed to just calling it? Also I wander if we should make it a little more clear what is happening and call this script Import-AzModules with even potentially pass in a version.

Reply:
I wasn't aware dot-source was different than just calling, I agree we can just call in this case.  I agree Import-AzModules is a better name, I just copied the name from their code.  I will make both these changes in the azure-sdk-tools PR.

I considered passing a version, but since we currently only use "latest" it seemed unnecessary.  Easy to add later if we want.

Reply:
https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_scripts?view=powershell-7#script-scope-and-dot-sourcing that is information about what happens if you put a "." in front of the script.